commit 7dd22c493b5ba6d7e4d731f6a7cbf0b28251d860 (refs/changes/66/2166/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-06-29T01:05:39+00:00 (2 years, 3 months ago)
    
    MANTA-3335 Muskie doesn't return after invoking callback with NotEnoughSpaceError
    Reviewed by: Jordan Hendricks <jordan.hendricks@joyent.com>
    Approved by: Jordan Hendricks <jordan.hendricks@joyent.com>

diff --git a/lib/picker.js b/lib/picker.js
index b2e797c..96f0f46 100644
--- a/lib/picker.js
+++ b/lib/picker.js
@@ -473,6 +473,7 @@ Picker.prototype.choose = function choose(opts, cb) {
 
         if (!sharks.length && (!tuple || tuple.length < replicas)) {
             cb(new NotEnoughSpaceError(size));
+            return;
         } else if (tuple && this.multiDC && replicas > 1) {
             var _dcs = tuple.map(function (s) {
                 return (s.datacenter);
